### DISCUSSION TITLE
Move the pause button on mobile

### DISCUSSION THREAD

**Discussion by WalmartWhiteCat** [ppy/osu#25461] [Ideas]
With the way Catch works, and now the new update to Mania, the gameplay field is overlapping with the pause button a lot, and with how these modes work, the hold activation time on the button is pretty much pointless seeing as you're giving an input for multiple seconds rather frequently to hit LN's or even just play Catch at all. It's far too easy to accidentally hit and is pretty annoying. I suggest moving to like, the upper corner or something.

(I really hope this isn't a duplicate, sorry if it is. I swear I searched for like, half an hour trying to find things similar, I just have no idea how to effectively use Github)

**Comment by yesseruser** [comment]
I agree. I'm having this issue as well and it does get annoying at times.

**Comment by yazsq** [comment]
yeah...playing !mania or !catch without that in the way would be much better

pause button on !catch:
![Screenshot_20241003_231924](https://github.com/user-attachments/assets/db32dc9e-7bc8-4908-9f32-892f8678ef6a)

pause button on !mania:
![Screenshot_20241003_230654](https://github.com/user-attachments/assets/cd8d86f7-18a0-4b58-a881-c6b84265564c)

**Reply by peppy** [reply]
All these overlay buttons makes it very hard to place it somewhere that doesn't interfere...

I guess we expose a way to place it per-ruleset?

**Reply by WalmartWhiteCat** [reply]
Wow, really didn't expect any response to this post, especially not almost a year later. I think a way to move the button would be best, yeah. Of course, moving it to a point where it's more difficult to accidentally hit would work fine, too.

I think the main problem is that the placement is almost exactly where many people would place their finger when holding a phone in their hands (near the centre edge). If it were on the upper edge of a touch area it would be less problematic since people probably aren't holding their phone in a way where hitting the top of a touch area is natural. Even if the pause button were lower it would be better, since that doesn't put it right in the way. Basically, as long as the button isn't where a person would naturally place their finger on an area (I.e. the middle), it would be much better.

But again, being able to move the button _would_ be the perfect solution for user experience.

**Reply by peppy** [reply]
I've seen other apps which allow dragging these buttons around at any point. Maybe this is the correct direction (and save the location globally)? The only downside or issue I can imagine with this would be if a user tries to drag a slider beneath the button they will drag the button along with the slider.

**Reply by yesseruser** [reply]
Maybe the button could be in the skin editor (and possibly forced to exist at least once)

**Reply by peppy** [reply]
Nope.

**Reply by WalmartWhiteCat** [reply]
> I've seen other apps which allow dragging these buttons around at any point. Maybe this is the correct direction (and save the location globally)? The only downside or issue I can imagine with this would be if a user tries to drag a slider beneath the button they will drag the button along with the slider.

Yeah, I could see that being possible. I think a good way to prevent that would be to make it so the button can only be moved in the pause menu. Then, add an option to the game that, when ticked on, will make it so the button maintains full visibility in the pause menu, but won't be interactable beyond just moving it. If that's at all possible, of course.

An alternative to that would be to make it unmovable while in the main game, but add an option to settings that opens a menu to adjust placement. The idea is, when the option is selected, it opens a blank canvas with simple blocks representing the playfield and skin elements (Minus health bar since it can cover the entire screen. Just put placeholder text on the canvas for that) for whatever mode is currently selected. Then the button can be moved around freely with minimal possibility for a miss input. After the user gets it how they want, they just leave the menu and they're done.

I'd say the first option is the greater of the two, but I can see both being viable in their own way. I hope I explained the ideas well enough.
